app/screens/home/search/search.tsx

```diff
@@ -281,7 +281,7 @@ const SearchScreen = ({teamId, teams}: Props) => {
     }, [isFocused, stateIndex]);
 
     const headerTopStyle = useAnimatedStyle(() => ({
-        top: lockValue.value ? lockValue.value : headerHeight.value,
+        top: lockValue || headerHeight.value,
```

Comment:
@larkox 

lockValue.value changes value by assigning a specific value during lock() and unlock() functions.

```ts
if (lock) {
  lockValue.value = defaultHeight;
}
```

and being used as a specific value:

```ts
const headerTopStyle = useAnimatedStyle(() => ({
  top: lockValue.value
  ...
});
```

But if the assignment was done like this, if we wanted a spring effect

```ts
if (lock) {
  lockValue.value = withSpring(defaultHeight);
}
```

then supplied to useAnimatedStyle like this:

```ts
const headerTopStyle = useAnimatedStyle(() => ({
  top: lockValue
  ...
});

<Animated.View style={headerTopStyle} />
```

then the above view will expand (or contract) it's top gradually with spring animation.  In theory.


However, in this case, the issue is caused by lockValue being a shared value and being included as part of the return object from useCollapsibleHeader hook. Check out demo below. Notice as I click on `click me`, my state variable increases by 20.  And animation is still moving properly.  However, the shared value lags behind.  This is because:

> When you change the sv.value the update will happen synchronously on the [UI thread](https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/glossary#ui-thread). On the other hand, on the [JavaScript thread](https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/glossary#javascript-thread) the update is asynchronous. This means when you try to immediately log the value after the change it will log the previously stored value.

Unfortunately, because lockValue as shared value mutates inside the hook as a direct assignment, and the value remain unchanged on JS, it doesn't update the lockValue within the component that calls the hook.

Hope this make sense.

https://github.com/mattermost/mattermost-mobile/assets/5550094/fac0a8a1-ab0c-4021-bd79-62b60b6f797c

Below is the code I use to create this simple demo:

```ts
    const isDemo = true;

    const [theValue, setTheValue] = useState(0);
    const sv = useSharedValue(0);

    const handlePress = () => {
        sv.value += 20;
        setTheValue((v) => v + 20);
    };

    const animatedStyles = useAnimatedStyle(() => ({
        transform: [{translateX: withSpring(theValue * 2)}],
    }), [theValue]);

    // uncomment to use shared value instead
    // const animatedStyles = useAnimatedStyle(() => ({
    //     transform: [{translateX: withSpring(sv.value * 2)}],
    // }), []);

    if (isDemo) {
        return (
            <View style={{flex: 1, justifyContent: 'center', alignItems: 'flex-start'}}>
                <Animated.View style={[styles.box, animatedStyles]}/>

                <View>
                    <Button
                        title='click me'
                        onPress={handlePress}
                    />
                    <Text>state: {theValue}</Text>
                    <Text>shared value: {sv.value}</Text>
                </View>
            </View>
        );
    }
```

Reply:
@enahum ^^^